database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -140,18 +140,32 @@ public void onCancelled(DatabaseError error) {
         return mDataSnapshots;
     }
 
-    private int getIndexForKey(String key) {
-        int dataCount = size();
-        int index = 0;
-        for (int keyIndex = 0; index < dataCount; keyIndex++) {
-            String superKey = mKeySnapshots.getObject(keyIndex);
-            if (key.equals(superKey)) {
-                break;
-            } else if (mDataSnapshots.get(index).getKey().equals(superKey)) {
-                index++;
+    private int returnOrFindIndexForKey(int index, String key) {
+        int realIndex;
```

Comment:
Should we initialize realIndex to 0 for clarity?  If  `key.equals(superKey)` is `true` on the first loop then we will break and return `realIndex` without ever having initialized it.

Reply:
Hmmm, we could but intellij complains because we do [initialize it to `dataIndex`](https://github.com/firebase/FirebaseUI-Android/pull/871/files/9dbd38b76cf88a795db270bc5f44b0dc11aab1b2#diff-848b2a3fc375b8ab8b5f22f22157614aR166) while out of the loop. Do you still want to proceed?

Reply:
Oh oops I missed that, thanks!